Simone on fhir must support

simoneOnFhir opened the pull request.

simoneOnFhir committed:
Best Oractice für Must-Support-Flags hinzugefügt

simoneOnFhir committed:
Update toc.yaml

patrick-werner commented on guides/Best-Practice-bei-der-Implementierung-und-Spezifizierung-mit-HL7/Übersicht/Spezifikation/Profilierung/Umgang-mit-Must-Support-Flags.page.md:
```diff
@@ -0,0 +1,24 @@
+* abgeleitete Profile können MustSupport-Flags nicht mehr entfernen, lediglich weitere Elemente kennzeichnen. Dies muss bei der Spezifikation bedacht werden, wenn es absehbar ist, dass abgeleitete Profile benötigt werden.
+* MustSupport Flags sollten auf allenen Ebenen der Baumstruktur konsistent verwendet werden. Das bedeutet dass
+  * ein Kind Element nur gekennzeichnet werden sollte, wenn auch das Parent-Element ein MustSupport-Flag hat.
+  * ein Parent-Element nur gegekennzeichnet werden sollte, wenn auch konkret spezifiziert werden kann, welche Kind-Elemente MustSupport sind.
```
```suggestion
  * ein Parent-Element nur gekennzeichnet werden sollte, wenn mindesten ein Kind-Element als MustSupport gekennzeichnet wird und konkret spezifiziert werden kann, welche Kind-Elemente MustSupport sind.  
```

simoneOnFhir committed:
Update guides/Best-Practice-bei-der-Implementierung-und-Spezifizierung-mit-HL7/Übersicht/Spezifikation/Profilierung/Umgang-mit-Must-Support-Flags.page.md

Co-authored-by: Patrick Werner <pa.f.werner@gmail.com>

mmreith commented on guides/Best-Practice-bei-der-Implementierung-und-Spezifizierung-mit-HL7/Übersicht/Spezifikation/Profilierung/Umgang-mit-Must-Support-Flags.page.md:
```diff
@@ -0,0 +1,24 @@
+Die Kennzeichnung relevanter Elemente mittels MustSupport im Sinne einer Positiv-Liste ist aus Interoperabilitätsgründen gegenüber dem Verbot von Elementen mittels Kardinalitäts-Begrenzung im Sinne einer einer Negativ-Liste vorzuziehen!
+
+Die Semantik von MustSupport ist im FHIR-Standard nicht erschöpfend definiert und kann an die Bedürfnisse einzelner Anwendungsfälle angepasst werden.
+Spezifikationen, die MustSupport-Flags verwenden, sind daher angehalten, die Erwartungen an die Impementierer im Leitfaden zu präzisieren.
```
Ist nicht eher eine Verpflichtung, da MS immer pro Spec definiert werden muss?

mmreith commented on guides/Best-Practice-bei-der-Implementierung-und-Spezifizierung-mit-HL7/Übersicht/Spezifikation/Profilierung/Umgang-mit-Must-Support-Flags.page.md:
```diff
@@ -0,0 +1,24 @@
+* MustSupport Flags sollten auf allenen Ebenen der Baumstruktur konsistent verwendet werden. Das bedeutet dass
+  * ein Kind Element nur gekennzeichnet werden sollte, wenn auch das Parent-Element ein MustSupport-Flag hat.
+  * ein Parent-Element nur gekennzeichnet werden sollte, wenn mindesten ein Kind-Element als MustSupport gekennzeichnet wird und konkret spezifiziert werden kann, welche Kind-Elemente MustSupport sind.  
+* der Grund, weshalb ein Element als MustSupport gekennzeichnet wurde, sollte möglichst in `ElementDefinition.comment` dargelegt werden.
```
Erzeugt das nicht einen großen Aufwand für die Spezifikateure und sollte dazu eigentlich im IG stehen und nicht in den Profilen? Auch um nicht alle Informationen in den Profilen zu haben?

alexzautke replied:
Unsere Empfehlung wäre den IG automatisiert auf Basis dieser Kommentare zu generieren. In unseren Projekten hatten wir das Problem, dass ansonsten Profil und IG von einander abweichen. Wenn ein Kommentar im Profil steht und man dann das Profil ändert stolpert man eher über den Kommentar und aktualisiert diesen.

simoneOnFhir replied:
Wir müssen auch immer bedenken, dass viele Implementierer die Profile als die Source of Truth sehen und gerade auf Simplifier sogar oft nur die Profile anschauen, nicht den IG...
Steng genommen geht es aber in der Empfehlung hier weniger um den ORT an dem man die Begründung festhält, sondern eher, DASS man es tut.

simoneOnFhir committed:
Enhance MustSupport flags section with Obligations

Added section on Obligations and clarified validation limitations for MustSupport flags.

mmreith commented on guides/Best-Practice-bei-der-Implementierung-und-Spezifizierung-mit-HL7/Übersicht/Spezifikation/Profilierung/Umgang-mit-Must-Support-Flags.page.md:
```diff
@@ -0,0 +1,30 @@
+{{page-title}}
+
+MustSupport-Flags dienen dazu, Implementierern die für einen konkreten UseCase relevanten Elemente aufzuzeigen.
+Die Kennzeichnung relevanter Elemente mittels MustSupport im Sinne einer Positiv-Liste ist aus Interoperabilitätsgründen gegenüber dem Verbot von Elementen mittels Kardinalitäts-Begrenzung im Sinne einer einer Negativ-Liste vorzuziehen!
```
@simoneOnFhir einer gibt es 2x

jschoen-gefyra replied:
```suggestion
Die Kennzeichnung relevanter Elemente mittels MustSupport im Sinne einer Positiv-Liste ist aus Interoperabilitätsgründen gegenüber dem Verbot von Elementen mittels Kardinalitäts-Begrenzung im Sinne einer Negativ-Liste vorzuziehen!
```

mmreith commented on guides/Best-Practice-bei-der-Implementierung-und-Spezifizierung-mit-HL7/Übersicht/Spezifikation/Profilierung/Umgang-mit-Must-Support-Flags.page.md:
```diff
@@ -0,0 +1,30 @@
+* der Grund, weshalb ein Element als MustSupport gekennzeichnet wurde, sollte möglichst in `ElementDefinition.comment` dargelegt werden.
+
+#### Obligations
+Wenn die Anforderungen an die relevanten Elemente in Implementierungen verschiedener Rollen (z.B. Sender und Empfänger) voneinander abweichen, bzw. abgeleiteten Profilen mehr Freiheiten gelassen werden sollen, Anforderungen für weitere Rollen hinzuzufügen, dann empfiehlt sich anstelle von (oder ergänzend zu) MustSupport-Flags die [Obligations-Extension](https://build.fhir.org/ig/HL7/fhir-extensio ...
```
@simoneOnFhir  Sollte man das wirklich schon bewerben, obwohl wir noch gar nicht genau wissen, ob oder wie gut das funktioniert?

jschoen-gefyra replied:
Wir haben letzte Woche im TC darüber gesprochen und gesagt, dass es kein bewerben, sondern ein Möglichkeiten aufzeigen ist. Aus dem Grund würden wir es drin lassen.

vwegert-adesso commented on guides/Best-Practice-bei-der-Implementierung-und-Spezifizierung-mit-HL7/Übersicht/Spezifikation/Profilierung/Umgang-mit-Must-Support-Flags.page.md:
```diff
@@ -0,0 +1,30 @@
+* abgeleitete Profile können MustSupport-Flags nicht mehr entfernen, lediglich weitere Elemente kennzeichnen. Dies muss bei der Spezifikation bedacht werden, wenn es absehbar ist, dass abgeleitete Profile benötigt werden.
+* MustSupport Flags sollten auf allenen Ebenen der Baumstruktur konsistent verwendet werden. Das bedeutet dass
+  * ein Kind Element nur gekennzeichnet werden sollte, wenn auch das Parent-Element ein MustSupport-Flag hat.
+  * ein Parent-Element nur gekennzeichnet werden sollte, wenn mindesten ein Kind-Element als MustSupport gekennzeichnet wird und konkret spezifiziert werden kann, welche Kind-Elemente MustSupport sind.  
```
So formuliert kann das leicht ausufern. Wenn ich eine MS simple extension habe, gibt doch der FHIR-Standard schon vor, dass value[x] angegeben werden muss (sonst wäre es keine simple extension). Wenn ich value[x] jetzt aufgrund dieser Vorgabe mit MS markiere, dann muss ich z.B. bei valueCoding nach der Logik auch code und system und potentiell weitere Elemente markieren. 
Wenn ich ein Period-Element als MS markiere - muss ich dann zwingend auch immer start und/oder end als MS markieren?
Das ufert halt schnell aus - und wenn nur aufgrund irgendwelcher Regeln/Best Practices der ganze Baum mit MS angepinselt wird, verliert das Kennzeichen schnell an Bedeutung.

jschoen-gefyra committed:
Update guides/Best-Practice-bei-der-Implementierung-und-Spezifizierung-mit-HL7/Übersicht/Spezifikation/Profilierung/Umgang-mit-Must-Support-Flags.page.md

jschoen-gefyra reviewed (approved):
Gemeinsam im TC am 28.5. besprochen und da keine Kommentare mehr eingegangen sind, mergen wir den jetzt. Weitere Diskussionen bitte auf Zulip dazu.
